extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlGet.java

```diff
@@ -0,0 +1,130 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.options.ListOptions;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+
+public class KubectlGet<ApiType extends KubernetesObject>
+    extends Kubectl.ApiClientBuilder<KubectlGet<ApiType>>
+    implements Kubectl.Executable<List<ApiType>> {
+
+  private String namespace;
+  private ListOptions listOptions;
+  private Class<ApiType> apiTypeClass;
+  private Class<? extends KubernetesListObject> apiTypeListClass;
+
+  KubectlGet(Class<ApiType> apiTypeClass) {
+    this.apiTypeClass = apiTypeClass;
+    this.listOptions = new ListOptions();
+  }
+
+  public KubectlGet<ApiType> apiListTypeClass(
+      Class<? extends KubernetesListObject> apiTypeListClass) {
+    this.apiTypeListClass = apiTypeListClass;
+    return this;
+  }
+
+  public KubectlGet<ApiType> options(ListOptions listOptions) {
+    this.listOptions = listOptions;
+    return this;
+  }
+
+  public KubectlGet<ApiType> namespace(String namespace) {
+    this.namespace = namespace;
+    return this;
+  }
+
+  public KubectlGetSingle name(String name) {
+    return new KubectlGetSingle(name);
+  }
+
+  @Override
+  public List<ApiType> execute() throws KubectlException {
+    GenericKubernetesApi<ApiType, ? extends KubernetesListObject> api =
+        apiTypeListClass == null
+            ? getGenericApi(apiTypeClass)
+            : getGenericApi(apiTypeClass, apiTypeListClass);
+    try {
+      if (isNamespaced()) {
+        return (List<ApiType>)
+            api.list(namespace, listOptions)
+                .onFailure(
+                    errorStatus -> {
+                      throw new ApiException(errorStatus.toString());
+                    })
+                .getObject()
+                .getItems();
+
+      } else {
+        return (List<ApiType>)
+            api.list(listOptions)
+                .onFailure(
+                    errorStatus -> {
+                      throw new ApiException(errorStatus.toString());
+                    })
+                .getObject()
+                .getItems();
+      }
+    } catch (ApiException e) {
+      throw new KubectlException(e);
+    }
+  }
+
+  private boolean isNamespaced() {
+    return !StringUtils.isEmpty(namespace);
+  }
+
+  public class KubectlGetSingle implements Kubectl.Executable<ApiType> {
```

Comment:
This is going to get messy/confusing to people because:
 
`Kubectl.get().namespace(foo).name(bar).execute()` will work, but `Kubectl.get().name(bar).namespace(foo)` won't.

I think we should either get rid of this class and merge the execute methods, or make this class extend from `KubectlGet`

Reply:
> Kubectl.get().namespace(foo).name(bar).execute() will work, but Kubectl.get().name(bar).namespace(foo) won't.

yeah i agree this can be confusing to the users. while the point here is that we're returning single object upon specifying resource name, and return a list of objects if the name absent. java doesn't support multi-return or overriding return type for the execute method  so merging or overriding the execute methods won't work.. 

in a word, if we're to make both `namespace(foo).name(bar)` and `name(bar).namespace(foo)` work, the execute method will have to return a list of object even when getting one object, which can be even more confusing i suppose?

Reply:
Hrm, why not just have, and then you can do:

`return new KubectlGetSingle().namespace(namespace).name(name)` in the function that adds the name, then it will work either way.

```
public KubectlGetSingle<ApiType> extends ResourceBuilder<...>, implements Kubectl.Executable<ApiType>

Reply:
done